### PR TITLE
Clickable asset links

### PR DESCRIPTION
**Pull request by yhpc**
- all asset types will now open non-empty urls, no url validation at this stage
- all asset types will now show URL in widget

### CODE REVIEW

File: lib/common.dart [hunk 1]
```diff
@@ -41,4 +43,23 @@ List<Jiffy> generateDays(Jiffy start, Jiffy end) {
   return list;
 }
 
+Future<void> openUrl(String url, BuildContext context) async {
```
Comment (kovarny) [inline lib/common.dart:46]: Don't think Future<void> is necessary.

File: lib/company/asset/abn_asset_widget.dart [hunk 2]
```diff
@@ -16,12 +17,14 @@ class ABNAssetWidget extends ConsumerWidget {
...
+  Widget build(BuildContext context, WidgetRef ref) => ref.watch(docSP(asset.path)).when(
             loading: () => Container(),
             error: (e, s) => ErrorWidget(e),
-            data: (assetDoc) => Text("${assetDoc.data()?['url'] ?? ''}")),
-        isThreeLine: true,
-      );
+            data: (assetDoc) => ListTile(
+              title: Text('ABN'),
+              subtitle: Text("${assetDoc.data()?['url'] ?? ''}"),
+              onTap: () => openUrl((assetDoc.data()?['url'] ?? ''), context), // in the future, can probably get rid of null handling
```
Comment (kovarny) [inline lib/company/asset/abn_asset_widget.dart:26]: Shouldn't be storing URL in ABN firstore doc. It should store just  the ABN.

File: lib/company/asset/facebook_asset_widget.dart [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+  FacebookAssetWidget(this.asset);
+
   @override
-  Widget build(BuildContext context, WidgetRef ref) => ListTile(
-      title: Text('Facebook'),
+  Widget build(BuildContext context, WidgetRef ref) => ref.watch(docSP(asset.path)).when(
+    loading: () => Container(), 
+    error: (e, s) => ErrorWidget(e), 
+    data: (assetDoc) => ListTile(
+      title: Text('Facebook - ${assetDoc.data()?["url"]}'),
```
Comment (kovarny) [inline lib/company/asset/facebook_asset_widget.dart:17]: Suggest we store ID s not the URLS

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/company/asset/abn_asset_widget.dart (hunk 2 above)
Comment (yhpc) [reply lib/company/asset/abn_asset_widget.dart]: Done in "Refacotr & Bug Fix"

File: lib/company/asset/facebook_asset_widget.dart (hunk 3 above)
Comment (yhpc) [reply lib/company/asset/facebook_asset_widget.dart]: Done in "Refacotr & Bug Fix"

**Review by kovarny — APPROVED** [review]

**Pull request merged**